packages/block-editor/src/components/link-control/index.js

```diff
@@ -146,22 +152,64 @@ function LinkControl( {
 								onChange( { ...value, ...suggestion } );
 							} }
 							isSelected={ index === selectedSuggestion }
-							isURL={ manualLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
+							isURL={ directLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
 							searchTerm={ inputValue }
 						/>
 					) ) }
+
+					{ showCreatePages && createEmptyPage && ! isInitialSuggestions && ! isSingleDirectEntryResult && (
+						<LinkControlSearchCreate
+							searchTerm={ inputValue }
+							onClick={ async () => {
+								setIsResolvingLink( true );
+								const newPage = await createEmptyPage( inputValue );
+								// TODO: handle error from API
+								setIsResolvingLink( false );
+								setIsEditingLink( false );
+								onChange( {
+									id: newPage.id,
+									title: newPage.title.raw, // TODO: use raw or rendered?
+									url: newPage.link,
+									type: newPage.type,
+								} );
+							} }
+						/>
+					) }
 				</div>
+
 			</div>
 		);
 	};
 
 	return (
 		<div className="block-editor-link-control">
-			{ ( ! isEditingLink ) && (
+
+			{ isResolvingLink && (
+				<div
+					className={ classnames( 'block-editor-link-control__search-item', {
+						'is-current': true,
+					} ) }
+				>
+					<span className="block-editor-link-control__search-item-header">
+						<span
+							className="block-editor-link-control__search-item-title"
+						>
+							{ __( 'Creating Page' ) }
+						</span>
+						<span className="block-editor-link-control__search-item-info">
+							{ __( 'Your new Page is being created' ) }.
```

Comment:
Again we're using `Page` here. Is there a way we can generalise this to allow other types of entities to be created by `LinkControl`? Would this be a config prop such as `createEntityName` which would be `page` or `post` or whatever entity the "Create" button creates?

Reply:
@shaunandrews Any ideas for much more generic "loading" label we could use? "Creating link…" or something like that which will work for pages and other types as well?

Reply:
Now provided https://github.com/WordPress/gutenberg/issues/18900#issuecomment-578200190

Reply:
@jeryj you completed this right? Feel free to `Resolve` if done.

Reply:
If it has to do with adding the loading indicator, then yes, it's done. The `block-editor-link-control__search-item-info` class looks like it's being used for something else now though.